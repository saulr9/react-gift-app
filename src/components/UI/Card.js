import React, { useState } from 'react';
import PropTypes, { objectOf } from 'prop-types';
import Avatar from './Avatar/Avatar';
import styles from './Card.module.css';

function Card({ img, title, user }) {
  const { display_name: userDisplayName, avatar_url: userAvatar } = user;
  const [loaded, setLoaded] = useState(false);
  const [hovered, setHovered] = useState(false);

  return (
    <div
      className={`${styles.card} ${!loaded && styles.card__loading} ${
        hovered && styles.card__hovered
      }`}
      onMouseEnter={() => setHovered(!hovered)}
      onMouseLeave={() => setHovered(!hovered)}
    >
      <div className={styles.card__images}>
        <img src={img} alt={title} onLoad={() => setLoaded(true)} />;
      </div>
      <div className={styles.card__details}>
        <Avatar img={userAvatar} username={userDisplayName} />
        <p>{title}</p>
      </div>
    </div>
  );
}
Card.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string,
  user: PropTypes.objectOf(PropTypes.any)
};
Card.defaultProps = {
  title: '',
  user: {
    display_name: 'No username',
    avatar_url:
      'https://gravatar.com/avatar/311ef4fa1fabdbc9b36d45f2ef906545?s=400&d=mp&r=x'
  }
};
export default Card;
