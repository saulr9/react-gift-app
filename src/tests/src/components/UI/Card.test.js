import { shallow } from 'enzyme';
import Card from '../../../../components/UI/Card';

describe('Testing UI/Card component', () => {
  const img = 'https://www.w3schools.com/w3css/img_lights.jpg';
  const title = 'goku';
  const user = {
    display_name: 'No username',
    avatar_url:
      'https://gravatar.com/avatar/311ef4fa1fabdbc9b36d45f2ef906545?s=400&d=mp&r=x'
  };
  const wrapper = shallow(<Card img={img} title={title} user={user} />);

  test('Should render Card Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a p with title goku', () => {
    const paragraph = wrapper.find('p');
    expect(paragraph.text().trim()).toBe(title);
  });

  test('img should have an alt text', () => {
    const image = wrapper.find('img');
    expect(image.prop('alt')).toBe(title);
  });

  test('img should have a url', () => {
    const image = wrapper.find('img');
    expect(image.prop('src')).toBe(img);
  });
});
