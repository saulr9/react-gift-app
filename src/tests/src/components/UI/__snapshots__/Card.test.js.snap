// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UI/Card component Should render Card Component 1`] = `
<div
  className="card card__loading false"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="card__images"
  >
    <img
      alt="goku"
      onLoad={[Function]}
      src="https://www.w3schools.com/w3css/img_lights.jpg"
    />
    ;
  </div>
  <div
    className="card__details"
  >
    <Avatar
      img="https://gravatar.com/avatar/311ef4fa1fabdbc9b36d45f2ef906545?s=400&d=mp&r=x"
      username="No username"
    />
    <p>
      goku
    </p>
  </div>
</div>
`;
